pi_ldapproxy/proxy.py

```diff
@@ -94,42 +102,50 @@ def authenticate_bind_request(self, request):
             # TODO: too much information revealed?
             result = (False, 'Could not determine realm.')
         else:
-            log.info('Resolved {dn!r} to {user!r}@{realm!r}', dn=request.dn, user=user, realm=realm)
+            log.info('Resolved {dn!r} to {user!r}@{realm!r} ({marker!r})',
+                     dn=request.dn, user=user, realm=realm, marker=app_marker)
             password = request.auth
-            response = yield self.request_validate(self.factory.validate_url,
-                                                   user,
-                                                   realm,
-                                                   password)
-            json_body = yield readBody(response)
-            if response.code == 200:
-                body = json.loads(json_body)
-                if body['result']['status']:
-                    if body['result']['value']:
-                        result = (True, '')
-                        # TODO: Is this the right place to bind the service user?
-                        if self.factory.bind_service_account:
-                            yield self.bind_service_account()
-                            self.bound = True
+            if self.factory.is_bind_cached(request.dn, app_marker, request.auth):
```

Comment:
This is potentially inefficient because it resolves the username in any case before checking if the ``(DN, app marker, password)`` tuple is contained in the bind cache.
But this is also a security feature if the user was removed in the meantime :)